ketcher-autotests/tests/Structure-Creating-&-Editing/Atom/Atom-Tool/group-generics-buttons-check.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { test } from '@playwright/test';
+import {
+  takeEditorScreenshot,
+  openFileAndAddToCanvas,
+  selectAtomInToolbar,
+  AtomButton,
+  clickOnAtom,
+  waitForPageInit,
+} from '@utils';
+
+test.describe('Generic nodes', () => {
+  test.beforeEach(async ({ page }) => {
+    await waitForPageInit(page);
+    await openFileAndAddToCanvas('Heteroatoms.mol', page);
+  });
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  test('G and G* adding to the atom of structure', async ({ page }) => {
+    /*
+        Test case: EPMLSOPKET-1503
+      */
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'G', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'O', 0);
+
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'GH', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'S', 0);
+
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'G*', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'F', 0);
+
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'GH*', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'I', 0);
+  });
+
+  test('Acylic atoms adding to the atom of structure', async ({ page }) => {
+    /*
+        Test case: EPMLSOPKET-1505
+      */
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'ACY', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'S', 0);
+
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'ACH', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'F', 0);
+  });
+
+  test('Acylic Carbo atoms adding to the atom of structure', async ({
+    page,
+  }) => {
+    /*
+        Test case: EPMLSOPKET-1508
+      */
+    await selectAtomInToolbar(AtomButton.Extended, page);
+    await page.getByRole('button', { name: 'ABC', exact: true }).click();
+    await page.getByTestId('OK').click();
+    await clickOnAtom(page, 'S', 0);
+
+    await selectAtomInToolbar(AtomButton.Extended, page);
```

Comment:
Can we create a separate function for these repetitive steps?

Reply:
done